x/name/wasm/encode.go

```diff
@@ -88,3 +98,9 @@ func (params *DeleteNameParams) Encode(contract sdk.AccAddress) ([]sdk.Msg, erro
 	msg := types.NewMsgDeleteNameRequest(record)
 	return []sdk.Msg{msg}, nil
 }
+
+// Encode creates a MsgCreateRootNameRequest.
+func (params *CreateRootNameParams) Encode(contract sdk.AccAddress) ([]sdk.Msg, error) {
+	msg := types.NewMsgCreateRootNameRequest(params.Authority, params.Name, params.Owner, params.Restricted)
+	return []sdk.Msg{msg}, nil
+}
```

Comment:
I suggest we don't include this in the wasm stuff.

I feel like, if we did want an encoder for this message, it would create the `MsgCreateRootNameRequest`, put it in a `MsgSubmitProposal` and return that.

I just don't trust that it's not possible for wasm to execute one of these messages arbitrarily.

While it might end up that we want to have this as an option in the long run, I'd like to err on the side of caution and plan out what these should look like first.

Reply:
I agree... I don't see a use case for a smart contract to generate a governance proposal.